### PR TITLE
Update release workflow

### PR DESCRIPTION
**Pull request by gastaldi**
- As needed in https://github.com/quarkiverse/quarkiverse-devops/issues/210

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -28,24 +28,19 @@ jobs:
...
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
           java-version: 17
           cache: 'maven'
           server-id: ossrh
           server-username: MAVEN_USERNAME
           server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE
```
Comment (turing85) [inline .github/workflows/release.yml:43]: Shouldn't this be a secret?

**Review by turing85 — COMMENTED** [review]
@gastaldi could you take a look? I lack the necessary information to see if the configuration is correct.

**Review by turing85 — APPROVED** [review]

**Comment by turing85** [comment]
@gastaldi should this change be downstreamed to other branches? I.e. `2.x` and `3.0.x`?

**Comment by gastaldi** [comment]
It depends on how long those branches will be active, that's up to you ;)

**Comment by turing85** [comment]
okay, this means yes :wink:

**Comment by turing85** [comment]
@gastaldi Sorry to pesker you. The cherry-pick was a bit tricky. Could you review:
- On branch `3.0.x`, commits:
  - [`3e6ebb6d913c70ed6f31a137234c75814a4d72d0`](https://github.com/quarkiverse/quarkus-artemis/commit/3e6ebb6d913c70ed6f31a137234c75814a4d72d0) 
  - [`bf96256c420202647b625e1f8de1528420082277`](https://github.com/quarkiverse/quarkus-artemis/commit/bf96256c420202647b625e1f8de1528420082277)
- On branch `2.x`, commit:
  - [`ef261e6ca4b60e93fd7000a5596e3cddceb6810c`](https://github.com/quarkiverse/quarkus-artemis/commit/ef261e6ca4b60e93fd7000a5596e3cddceb6810c)

This would be much appreciated :slightly_smiling_face:

**Comment by gastaldi** [comment]
LGTM 👍🏻

**Comment by turing85** [comment]
Thanks!

**Pull request merged**